src/track/beatgrid.cpp

```diff
@@ -184,13 +190,49 @@ double BeatGrid::findNthBeat(double dSamples, int n) const {
         n = n + 1;
     }
 
-    double dResult = dClosestBeat + n * m_dBeatLength;
+    double dResult = floor(dClosestBeat + n * m_dBeatLength);
     if (!even(static_cast<int>(dResult))) {
         dResult--;
     }
     return dResult;
 }
 
+QPair<double, double> BeatGrid::findPrevNextBeats(double dSamples) const {
+    QMutexLocker locker(&m_mutex);
```

Comment:
Isn't a breatgrid constant during the lifetime of a track? Why do we need to lock at all?